src/Http/Http.Extensions/src/RequestDelegateFactoryOptions.cs

```diff
@@ -36,4 +36,14 @@ public sealed class RequestDelegateFactoryOptions
     /// The list of filters that must run in the pipeline for a given route handler.
     /// </summary>
     public IReadOnlyList<Func<RouteHandlerContext, RouteHandlerFilterDelegate, RouteHandlerFilterDelegate>>? RouteHandlerFilterFactories { get; init; }
+
+    /// <summary>
+    /// The default endpoint metadata to add as part of the creation of the <see cref="RequestDelegateResult.RequestDelegate"/>.
+    /// </summary>
+    /// <remarks>
+    /// This metadata will be included in <see cref="RequestDelegateResult.EndpointMetadata" /> after any metadata inferred during creation of the
+    /// <see cref="RequestDelegateResult.RequestDelegate"/> but before any metadata provided by types in the delegate signature that implement
+    /// <see cref="IEndpointMetadataProvider" /> or <see cref="IEndpointParameterMetadataProvider" />.
```

Comment:
Why isn't this the following?

```suggestion
    /// This metadata will be included in <see cref="RequestDelegateResult.EndpointMetadata" /> before any metadata inferred during creation of the
    /// <see cref="RequestDelegateResult.RequestDelegate"/> and before any metadata provided by types in the delegate signature that implement
    /// <see cref="IEndpointMetadataProvider" /> or <see cref="IEndpointParameterMetadataProvider" />.
```

I see that it's currently implemented as described, but wouldn't it be more logical to put all the metadata inferred by RDF and inferred by MetadataProviders together after the `DefaultEndpointMetadata`.